### PR TITLE
[TS-408] Forgot Password!

### PR DESCRIPTION
**Pull request by GadoliniumPotassium**
You are not able to get there from the forgot password router because for some reason you have to be logged in to see it but if you log in and have /forgotpasswordemail and /forgotpassword/userId=something you will get there!

### CODE REVIEW

File: src/pages/ForgotPassword/ForgotPasswordEmail.vue [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  name: 'EmailSender',
+  data: () => ({
+    valid: '',
+    email: '',
+    rules: {
+      required: v => !!v || 'Required'
+    },
+    rulesEmail: {
+      format: v =>
+        /^(([^<>()[\]\\.,;:\s@']+(\.[^<>()\\[\]\\.,;:\s@']+)*)|('.+'))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(
```
Comment (bowe99) [inline src/pages/ForgotPassword/ForgotPasswordEmail.vue:55]: Can you switch out all the places where you use [0-9] to \d and you have a couple more code smells to fix

**Review by bowe99 — CHANGES_REQUESTED** [review]

File: src/pages/ForgotPassword/ForgotPasswordEmail.vue (hunk 1 above)
Comment (GadoliniumPotassium) [reply src/pages/ForgotPassword/ForgotPasswordEmail.vue]: I will change what I can but I can't do anything about the regex being too big when its literally the same regex from the sign up page.

File: src/pages/ForgotPassword/ForgotPasswordEmail.vue [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    async submit () {
+      if (this.$refs.emailForm.validate()) {
+        try {
+          const UserInfo = await axiosInstance.get(`/userInfo?searchQuery=${this.email}&limit=1`)
+          const MailOption = {
+            to: this.email,
+            subject: 'TradeShare: Forgot Password',
+            text: `Change your password in the following link http://localhost:8081/forgotpassword/userId=${UserInfo.data.data[0].userId}`
+          }
+          print('sending MailOption to mailer')
```
Comment (Ming424) [inline src/pages/ForgotPassword/ForgotPasswordEmail.vue:70]: image
too much python

**Comment by Ming424** [comment]
image
I can get email but url redirect to login page
image

**Comment by GadoliniumPotassium** [comment]
I just noticed its because of a typo, I'm fixing it now!

**Review by tungleu — APPROVED** [review]

**Review by Ming424 — APPROVED** [review]

**Pull request merged**